### DISCUSSION TITLE
failed to calculate checksum of ref

### DISCUSSION THREAD

**Discussion by netsurion-jsayre** [dotnet/dotnet-docker#4716] [Q&A] [answered]
New to Docker. Using Ubuntu 20.04.6LTS. Using docker file "Dockerfile.ubuntu" and ".dockerignore" file in my directory. I realize that the provided Dockerfile is for 22.04, and am hoping that's not my issue. Anyway, getting below error.

`ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref TJDJ:TJG5:3FRL:MTEA:NIET:UVVI:MRJU:XFAX:XLC5:GUZE:2ZNY:WOCC::hns4t44fjx7oot0q2zc7ymhcz: "/aspnetapp": not found`

This error is received when execute `sudo docker build --force-rm=true --no-cache=true .` and variations (tagging, not using no-cache, etc.). If I `sudo docker run` the sample image, I have no issues. I have also tried executing the build from a higher-level directory and specifying the directory of the `Dockerfile`, but that has not worked either. Google-fu has failed, ChatGPT suggested I restart docker (haha), and I haven't seen anything else that really seems similar. Thoughts? Below my sig are my `Dockerfile` and `.dockerignore`. Sorry for lousy formatting, I always seem to jack that up as I don't do this much. Thanks in advance.
-jcs

**Dockerfile**
```
# Learn about building .NET container images:
# https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md
FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy AS build
WORKDIR /source

# copy csproj and restore as distinct layers
COPY aspnetapp/*.csproj .
RUN dotnet restore --use-current-runtime

# copy everything else and build app
COPY aspnetapp/. .
RUN dotnet publish --use-current-runtime --self-contained false --no-restore -o /app

# final stage/image
FROM mcr.microsoft.com/dotnet/aspnet:7.0-jammy
WORKDIR /app
COPY --from=build /app .
ENTRYPOINT ["./aspnetapp"]
```

**.dockerignore**
```
# directories
**/bin/
**/obj/
**/out/

# files
Dockerfile*
**.*.md
```

**Comment by mthalman** [comment, marked as answer]
What is your current working directory when executing the `docker build` command? That is the directory that you are setting as your [build context](https://docs.docker.com/build/building/context/), via the `.` character at the very end of your command arguments. And that means any `COPY` instructions in your Dockerfile are relative to that directory. So that directory should contain the `aspnetapp` directory that you reference in `COPY aspnetapp/*.csproj .`. If you want to specify a different directory as your build context, you should replace the `.` in your `docker build` command with the path that you do want.

**Reply by netsurion-jsayre** [reply]
Well, that solved that problem, thanks a ton. But it introduced a new one. Screenshot below. I will try to work through it on my own and mark this as the answer. Thanks again!

![image](https://github.com/dotnet/dotnet-docker/assets/49252806/7babe4dd-fb27-453a-b1fd-79700986ef64)
